synfig-studio/src/gui/trees/layertreestore.cpp

```diff
@@ -199,8 +199,10 @@ LayerTreeStore::get_value_vfunc(const Gtk::TreeModel::iterator& iter, int column
 			if (column == model.index.index())
 				set_gvalue_tpl<int>(value, layer->get_depth());
 			else
-			if (column == model.z_depth.index())
-				set_gvalue_tpl<float>(value, layer->get_true_z_depth(canvas_interface()->get_time()));
+			if (column == model.z_depth.index()){
+				Glib::ustring str = remove_trailing_zeroes(std::to_string(layer->get_true_z_depth(canvas_interface()->get_time())));
+				set_gvalue_tpl<Glib::ustring>(value, str);//the value itself doesnt have trailing zeroes but seems like the renderer does it
```

Comment:
> the value itself doesnt have trailing zeroes but seems like the renderer does it

Please check `std::to_string` documentation.

Reply:
oh yes, I did check it. this was an old comment before I switched the float into a string, but i'm guessing the same still applies. the CellRendererText renders the values as some form of string which is probably why this all happens.